static/app/utils/useTimeout.tsx

```diff
@@ -17,35 +17,46 @@ function useTimeout({timeMs, onTimeout}: Options) {
     timeoutRef.current = timeout;
   }, []);
 
+  const start = useCallback(() => {
+    saveTimeout(null);
+    saveTimeout(setTimeout(onTimeout, timeMs));
+  }, [onTimeout, saveTimeout, timeMs]);
+
+  const cancel = useCallback(() => {
+    saveTimeout(null);
+  }, [saveTimeout]);
+
+  const end = useCallback(() => {
+    saveTimeout(null);
+    onTimeout();
+  }, [onTimeout, saveTimeout]);
+
+  // Cancel the timeout on unmount
+  useEffect(() => {
+    return () => {
+      cancel();
+    };
```

Comment:
would it still work without the extra fn wrapper?

```suggestion
    return cancel;
    };
```

Reply:
Yeah I always wrap so it's clear, but I went ahead and removed it